v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/exceptions/ChangeEventTypeConvertorException.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.exceptions;
```

Comment:
Initially `ChangeEventConvertorException` was used for everything - errors in `ChangeEventTypeConvertor` as well as every other place. 

We have only moved the `ChangeEventTypeConvertor` to a common folder, the rest is streaming template specific. Hence, added a new type of exception called `ChangeEventConvertorException` for error to the common file only.

Alternative to this would be keeping the same original exception `ChangeEventConvertorException` for everything and move that exception to the common folder. But then the structuring feels a little haphazard:
```
|_ datastream-to-spanner - contains changeEventConvertor
|_ spanner-common - contains changeEventConvertorException and changeEventTypeConvertor
```

Reply:
OR we can rename the name to a generic one - `conversionException` and use it for bulk as well.

Reply:
What is the difference between the two exceptions ? We should define the purpose of each up front and use it accordingly.

Reply:
ChangeEventTypeConvertorException is for exceptions during data conversion only, like int to string, string to timestamp etc.
ChangeEventConvertorException is for any other error in the conversion process, like building shadow table mutations, verification failed because table does not exist in spanner, unsupported columnType etc.

Reply:
Removed 2 separate exceptions, kept the original exception structure and moved it to the common package.